src/sagemaker/predictor.py

```diff
@@ -114,18 +114,35 @@ def _delete_endpoint_config(self):
         self.sagemaker_session.delete_endpoint_config(endpoint_config_name)
 
     def delete_endpoint(self, delete_endpoint_config=True):
-        """Delete the Amazon SageMaker endpoint and endpoint configuration backing this predictor.
+        """Delete the Amazon SageMaker endpoint backing this predictor. Also delete the endpoint configuration attached
+           to it if delete_endpoint_config is True.
 
         Args:
-            delete_endpoint_config (bool): Flag to indicate whether to delete the corresponding SageMaker endpoint
-                configuration tied to the endpoint. If False, only the endpoint will be deleted. (default: True)
+            delete_endpoint_config (bool): Flag to indicate whether to delete endpoint configuration together with
+                endpoint. If False, only endpoint will be deleted. Default: True.
```

Comment:
Let's follow this for our docstrings.

https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html